mkosi/qemu.py

```diff
@@ -646,20 +646,62 @@ def finalize_firmware_variables(
     stack: contextlib.ExitStack,
 ) -> tuple[Path, str]:
     ovmf_vars = stack.enter_context(tempfile.NamedTemporaryFile(prefix="mkosi-ovmf-vars-"))
+    ovmf_json = stack.enter_context(tempfile.NamedTemporaryFile(prefix="mkosi-ovmf-vars-json-"))
+    have_mok_json = False
+
+    if config.secure_boot_certificate:
+        try:
+            from virt.firmware.efi import guids, siglist
+
+            # In order to make the kernel use mkosi.crt to verify dm-verity volumes, we need to
+            # create runtime and volatile EFI variables, MokListTrustedRT and MokListRT, as
+            # it's only from kernel 6.11 that db is used for this purpose. Note that, unlike
+            # certificates in db, the certificates in MOK are only used if secure boot is enabled.
+            # First create an efivar with the certificate, and then read it and convert to the
+            # json format that virt-fw-vars expects.
+            # attr=4 means EFI_VARIABLE_RUNTIME_ACCESS, and the GUID is the well-known MOK one.
+
+            mok_sigdb = siglist.EfiSigDB()
+            mok_sigdb.add_cert(guids.parse_str("605dab50-e046-4300-abb6-3dd810dd8b23"), config.secure_boot_certificate)
+            mok_json = dict(
+                version=2,
+                variables=[
+                    dict(
+                        name="MokListRT",
+                        guid="605dab50-e046-4300-abb6-3dd810dd8b23",
+                        attr=4,
+                        data=bytes(mok_sigdb).hex(),
+                    ),
+                    dict(
+                        name="MokListTrustedRT",
+                        guid="605dab50-e046-4300-abb6-3dd810dd8b23",
+                        attr=4,
+                        data="01",
+                    ),
+                ],
+            )
```

Comment:
For the environment variables I do understand where wanting to use `dict` comes from, but let's not make this too obscure
```suggestion
            mok_json = {
                "version": 2,
                "variable" [
                    {
                        "name": "MokListRT",
                        "guid": "605dab50-e046-4300-abb6-3dd810dd8b23",
                        "attr": 4,
                        "data": bytes(mok_sigdb).hex(),
                    },
                    {
                        "name": "MokListTrustedRT",
                        "guid": "605dab50-e046-4300-abb6-3dd810dd8b23",
                        "attr": 4,
                        "data": "01",
                    },
                ],
            }
```